test/e2e/specs/widgets/editing-widgets.spec.js

```diff
@@ -0,0 +1,723 @@
+/**
+ * WordPress dependencies
+ */
+const {
+	test: base,
+	expect,
+} = require( '@wordpress/e2e-test-utils-playwright' );
+
+/** @typedef {import('@wordpress/e2e-test-utils-playwright').Editor} Editor */
+/** @typedef {import('@playwright/test').Locator} Locator */
+
+/** @type {ReturnType<typeof base.extend<{widgetsScreen: WidgetsScreen}>>} */
+const test = base.extend( {
+	widgetsScreen: async ( { page, editor }, use ) => {
+		await use( new WidgetsScreen( { page, editor } ) );
+	},
+} );
+
+test.describe( 'Widgets screen', () => {
+	test.beforeEach( async ( { admin, editor } ) => {
+		await admin.visitAdminPage( 'widgets.php' );
+
+		await editor.setPreferences( 'core/edit-widgets', {
+			welcomeGuide: false,
+		} );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllWidgets();
+	} );
+
+	test.beforeAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			// TODO: Ideally we can bundle our test theme directly in the repo.
+			requestUtils.activateTheme( 'twentytwenty' ),
+			requestUtils.deleteAllWidgets(),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'Should insert content using the global inserter', async ( {
+		page,
+		pageUtils,
+		widgetsScreen,
+	} ) => {
+		await expect(
+			widgetsScreen.updateButton,
+			'Update button should start out disabled'
+		).toBeDisabled();
+
+		const [ firstWidgetArea, secondWidgetArea ] =
+			await widgetsScreen.widgetAreas.all();
+
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
```

Comment:
Nit (not a blocker): Should we make the locator more specific?

```
await page
			.getByRole( 'toolbar', { name: 'Document tools' } )
			.getByRole( 'button', { name: 'Toggle block inserter' } )
			.click();
```

Reply:
Sure!